### PR TITLE
Separate readmap R script to file

### PR DESCRIPTION
**Pull request by mvdbeek**
match requirements to available conda packages and update gridExtra to current
syntax.

### CODE REVIEW

File: tools/msp_sr_readmap_and_size_histograms/readmap.xml [hunk 1]
```diff
@@ -1,239 +1,109 @@
...
-		  --normalization_factor
-		  #for $i in $refGenomeSource.series
-    		    $i.norm
-		  #end for
-		  #if $gff:
-		    --gff
-                    $gff
-                  #end if
-
+<tool id="Readmap" name="Generate readmap and histograms from alignment files" version="1.2.0">
```
Comment (drosofff) [inline tools/msp_sr_readmap_and_size_histograms/readmap.xml]: Yes, but after my rallye for upgrading Oases in metavisitor 1.2, I will not do it again before a while. If you bump msp_sr_readmap_and_size_histograms, which would be nice, I would like you to ensure tool/workflow/docker update process ;-)

**Comment by drosofff** [comment]
Yes, see my line note ;-)

File: tools/msp_sr_readmap_and_size_histograms/test-data/Readmap_dataframe.tab [hunk 2]
```diff
@@ -2373,13 +2547,21 @@ FBti0020401	5811	17.0	F	sample3.srbowtie_out
 FBti0020401	5996	-1.0	R	sample3.srbowtie_out
 FBti0020401	6183	-1.0	R	sample3.srbowtie_out
 FBti0020401	6218	1.0	F	sample3.srbowtie_out
+FBti0020401	6348	0	F	sample3.srbowtie_out
+FBti0020406	0	0	F	sample3.srbowtie_out
 FBti0020406	161	1.0	F	sample3.srbowtie_out
 FBti0020406	697	-1.0	R	sample3.srbowtie_out
+FBti0020406	812	0	F	sample3.srbowtie_out
+FBti0019511	0	0	F	sample3.srbowtie_out
```
Comment (mvdbeek) [inline tools/msp_sr_readmap_and_size_histograms/test-data/Readmap_dataframe.tab]: It seems there is still something wrong here, this should have been skipped :/

Comment (mvdbeek) [inline tools/msp_sr_readmap_and_size_histograms/test-data/Readmap_dataframe.tab]: Ah, yes, it is correct after all:
https://github.com/ARTbio/tools-artbio/pull/79/files#diff-2a5423b1092f0853a89d34f45bc3ee65R1335

**Comment by mvdbeek** [comment]
@drosofff can we get this in? This works well from the testtoolshed and will allow us to use conda for this tool.

**Comment by mvdbeek** [comment]
and yes, I will manage the checklist for tool upgrades.

**Comment by drosofff** [comment]
Of course Yes.
I am a bit confused, a couple of days ago, I looked at the testtoolshed: both msp_sr_readmap_and_size_histograms and msp_sr_size_histograms did not change since April.
What works well in the testtoolshed specifically ?
And also for my formation: what in the tool codes makes it conda complient ? Is biocbasics packaged and available somewhere (anaconda ?).
Sorry for my messy questions, I doing many things in parallel these days.

But yes, we can merge


> This works well from the testtoolshed and will allow us to use conda for this tool.

**Comment by mvdbeek** [comment]
> both msp_sr_readmap_and_size_histograms and msp_sr_size_histograms did not change since April.

Yes, this PR dates from April, but conda wasn't quite ready then.

> What works well in the testtoolshed specifically ?
msp_sr_readmap, the whole functionality. Tested with planemo and cloud instances.

> what in the tool codes makes it conda complient ? Is biocbasics packaged and available somewhere

In the requirements tag we are specifying the packages that we need for the plotting, which are
 - r-gridextra
 - r-latticeextra
 - r-optparse

These contain all the imports we're using in R.
The majority of the code changes deal with changes in the gridextra package that has changed sometime after R version 3.1.2.
There is no biocbasics equivalent (and it has always been messy to have this huge blob of software, where you have no idea what's inside).

And because I find that developing R scripts and debugging them when they are in the <configfiles> section of a galaxy tool is difficult, I made it a standalone script (`plot_size_readmap.r`).

> Sorry for my messy questions, I doing many things in parallel these days.

No worries, let me know if this is clear now

**Comment by drosofff** [comment]
Yes I see now that biocbasic vanished. 👍 
let me test in a workflow before merging. But.... Does it imply that conda functionalities have to be activated in galaxy.ini as well as the priorities in dependency_resolvers_conf.xml ?

**Comment by mvdbeek** [comment]
> Does it imply that conda functionalities have to be activated in galaxy.ini as well as the priorities in dependency_resolvers_conf.xml ?

As long as you use release 16.07, you just have to leave the `install conda dependencies` checkbox checked when you install a tool. But if you merge this I will come back with workflow results and open a PR against ansible-artimed.

**Comment by drosofff** [comment]
> As long as you use release 16.07
@mvdbeek Today I am rephrasing my question  (Does it imply that conda functionalities have to be activated ?)

is this tool version working on 16.04 ?

Cause, on mississippi, I have the error in https://mississippi.snv.jussieu.fr/u/drosofff/h/test-readmap-12

**Comment by mvdbeek** [comment]
If you're using 16.04 and don't want to upgrade to 16.07, you have to be on the latest commit in the branch and either activate `conda_auto_install` (which is a bit annoying, because it tries to install dependencies everytime a dependency can't be resolved) and `conda_auto_init` or install the conda dependencies manually using
```
conda create --name __bowtie@1.0.0 bowtie==1.0.0 --channel bioconda
conda create --name __pysam@0.9.0 pysam==0.9.0 --channel bioconda
conda create --name numpy@1.9.3 numpy==1.9.3 --channel bioconda
conda create --name  r-optparse@1.3.0 r-optparse==1.3.0 --channel bioconda
conda create --name r-latticeextra@0.6_26 r-latticeextra==0.6_26 --channel bioconda
conda create --name r-gridextra@2.0.0 r-gridextra==2.0.0 --channel bioconda
```
For the different conda options you can see this document:
https://github.com/galaxyproject/galaxy/blob/dev/doc/source/admin/conda_faq.rst#1-how-do-i-enable-conda-dependency-resolution-for-galaxy-jobs

**Pull request merged**